src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -294,6 +314,49 @@ internal Silo(string name, SiloType siloType, ClusterConfiguration config, ILoca
             logger.Info(ErrorCode.SiloInitializingFinished, "-------------- Started silo {0}, ConsistentHashCode {1:X} --------------", SiloAddress.ToLongString(), SiloAddress.GetConsistentHashCode());
         }
 
+        private void InitializeDependencyResolver()
+        {
+            if (String.IsNullOrWhiteSpace(nodeConfig.DependencyResolverProviderName))
+            {
+                return;
+            }
+
+            //
+            // Load the assemblies from the Silo's directory, since the dependency resolver's type is not directly referenced,
+            // so the ProviderLoader will not find it within the loaded assemblies.
+            //
+
+            var siloRoot = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+            var directories = new Dictionary<string, SearchOption>
+            {
+                {siloRoot, SearchOption.TopDirectoryOnly}
+            };
+
+            AssemblyLoaderReflectionCriterion[] loadCriteria =
+            {
+                AssemblyLoaderCriteria.LoadTypesAssignableFrom(
+                    typeof (IDependencyResolverProvider))
+            };
+
+            var discoveredAssemblyLocations = AssemblyLoader.LoadAssemblies(directories, null, loadCriteria, logger);
+
+            var dependencyResolverProviderManager = new DependencyResolverProviderManager(ProviderCategoryConfiguration.DEPDENDENCY_RESOLVER_PROVIDER_CATEGORY_NAME);
```

Comment:
can all the code in this method go into `DependencyResolverProviderManager`, so here we will only instantiate `DependencyResolverProviderManager`, call Load on it and then call `GetDependencyResolver` or something? I just don't want to pollute the silo code with all those details about `AssemblyLoaderReflectionCriterion` and etc...